pkgs/tools/misc/shim/default.nix

```diff
@@ -0,0 +1,55 @@
+{ stdenv, fetchFromGitHub, lib, elfutils, vendorCertFile ? null
+, defaultLoader ? null, ... }:
+
+let
+  gnu-efi-src = fetchFromGitHub {
+    owner = "rhboot";
+    repo = "gnu-efi";
+    rev = "03670e14f263ad571bf0f39dffa9b8d23535f4d3";
+    hash = "sha256-iw+cwaceMaHILmbQXVXOFxlsLWGv0M+x7aVzc8puTDA=";
+  };
+
+  inherit (stdenv.targetPlatform) system;
+  throwSystem = throw "Unsupported system: ${system}";
+
+  target = {
+    x86_64-linux = "shimx64.efi";
+    aarch64-linux = "shimaa64.efi";
+  }.${system} or throwSystem;
+in stdenv.mkDerivation rec {
+  pname = "shim";
+  version = "15.7";
+
+  src = fetchFromGitHub {
+    owner = "rhboot";
+    repo = pname;
+    rev = version;
+    hash = "sha256-TRSpyZC4pt1wU3pxQ0WV6ARpPWPd3qwiJCtilzSLLXQ=";
+  };
+
+  postPatch = ''
+    cp -a ${gnu-efi-src}/* gnu-efi/
+  '';
+
+  buildInputs = [ elfutils ];
+
+  NIX_CFLAGS_COMPILE = [ "-I${toString elfutils.dev}/include" ];
+
+  makeFlags =
+    lib.optional (vendorCertFile != null) "VENDOR_CERT_FILE=${vendorCertFile}"
+    ++ lib.optional (defaultLoader != null) "DEFAULT_LOADER=${defaultLoader}"
+    ++ [ target ];
+
+  installPhase = ''
+    mkdir $out
+    cp ${target} $out/
```

Comment:
`$out/bin` ?

Reply:
it's not an elf you could execute

Reply:
hm agreed, but I'm not sure, it's an enough reason but no strong feeling

Reply:
Other code places this usually in `$out/share/<program>/..`. But having it in `$out` is probably also fine.